From: Jeff Mahoney <jeffm@suse.com>
Subject: configfs: API compatibility layer

 ConfigFS makes use of a different kref API, but provides a compatibility
 layer for use until the kref API matches the one it uses.

 fs/configfs/Makefile                        |    2 ++
 fs/configfs/kapi-compat/include/kref_init.h |    9 +++++++++
 2 files changed, 11 insertions(+)
diff -ruNpX dontdiff linux-2.6.14/fs/configfs/kapi-compat/include/kref_init.h linux-2.6.14.ocfs2/fs/configfs/kapi-compat/include/kref_init.h
--- linux-2.6.14/fs/configfs/kapi-compat/include/kref_init.h	1969-12-31 19:00:00.000000000 -0500
+++ linux-2.6.14.ocfs2/fs/configfs/kapi-compat/include/kref_init.h	2005-11-01 11:24:33.000000000 -0500
@@ -0,0 +1,9 @@
+#ifndef KAPI_KREF_INIT_H
+#define KAPI_KREF_INIT_H
+
+#include <linux/kref.h>
+
+/* kernel folks 1, universe 0 */
+#define kref_init(kref_obj, release_func)	kref_init(kref_obj)
+
+#endif /* KAPI_KREF_INIT_H */
diff -ruNpX dontdiff linux-2.6.14/fs/configfs/Makefile linux-2.6.14.ocfs2/fs/configfs/Makefile
--- linux-2.6.14/fs/configfs/Makefile	1969-12-31 19:00:00.000000000 -0500
+++ linux-2.6.14.ocfs2/fs/configfs/Makefile	2005-11-01 11:24:54.000000000 -0500
@@ -1,7 +1,9 @@
 #
 # Makefile for the configfs virtual filesystem
 #
 
 obj-$(CONFIG_CONFIGFS_FS)      += configfs.o
+
+EXTRA_CFLAGS += -include fs/configfs/kapi-compat/include/kref_init.h
 
 configfs-objs  := inode.o file.o dir.o symlink.o mount.o item.o
